railgun/manager/manager.go

```diff
@@ -278,13 +280,60 @@ func Run(ctx context.Context, c *Config) error {
 		return err
 	}
 
+	// build the kongConfig
 	kongConfig := sendconfig.Kong{
 		URL:         c.KongAdminURL,
 		FilterTags:  []string{c.FilterTag},
 		Concurrency: c.Concurrency,
 		Client:      kongClient,
 	}
 
+	// download the kong root configuration (and validate connectivity with the proxy)
+	setupLog.Info("downloading the kong root configuration from the proxy")
+	root, err := kongClient.Root(ctx)
+	if err != nil {
+		setupLog.Error(err, "could not download kong root configuration")
+		return err
+	}
+	proxyConfig, ok := root["configuration"].(map[string]interface{})
+	if !ok {
+		err := fmt.Errorf("invalid root configuration, expected map[string]interface{} got %t", root["configuration"])
+		setupLog.Error(err, "could not determine proxy configuration")
+		return err
+	}
+	proxyDB, ok := proxyConfig["database"].(string)
+	if !ok {
+		err := fmt.Errorf("invalid database configuration, expected a string got %t", proxyConfig["database"])
+		setupLog.Error(err, "could not determine proxy database configuration")
+		return err
+	}
+
+	// validate the kong database configuration
+	if proxyDB == "cassandra" {
+		err := fmt.Errorf("Cassandra-backed deployments of Kong managed by the ingress controller are no longer supported; you must migrate to a Postgres-backed or DB-less deployment")
+		setupLog.Error(err, "unable to integrate with upstream Kong Proxy")
+		return err
+	}
+	if proxyDB == "off" {
+		kongConfig.InMemory = true
+	}
+
+	// capture the Kong Proxy version
+	proxyStringVersion, ok := root["version"].(string)
+	if !ok {
+		err := fmt.Errorf("invalid value found for version in kong root config: %t", root["version"])
+		setupLog.Error(err, "unable to determine Kong proxy version")
+		return err
+	}
+	proxySemver, err := ctrlutils.GetSemver(proxyStringVersion)
+	if err != nil {
+		setupLog.Error(err, "could not parse proxy version %s as semver", proxyStringVersion)
+		return err
+	}
+	setupLog.Info("found kong proxy server version %s", proxySemver)
+	kongConfig.Version = proxySemver
```

Comment:
:+1: I'll convert

Reply:
It looks like this was merged yesterday and hasn't yet made it into a release, rather than hold this PR up for a go-kong release I've spun off an issue:

https://github.com/Kong/kubernetes-ingress-controller/issues/1315

Reply:
Let's do it right away to avoid introduction of debt - I've just cut go-kong v0.19.0 with the required capabilities.

Reply:
Ok, thanks for cutting the release. The patch for this is 8ef945dacdb61d0e6e4ce65c3a435f9f7519b6e0 :+1: